lib/web/integrations_awsoidc.go

```diff
@@ -270,11 +270,15 @@ func (h *Handler) awsOIDCConfigureDeployServiceIAM(w http.ResponseWriter, r *htt
 	}
 
 	// Ensure the IntegrationName is valid.
-	_, err = h.GetProxyClient().GetIntegration(ctx, integrationName)
+	integration, err := h.GetProxyClient().GetIntegration(ctx, integrationName)
 	// NotFound error is ignored to prevent disclosure of whether the integration exists in a public/no-auth endpoint.
 	if err != nil && !trace.IsNotFound(err) {
 		return nil, trace.Wrap(err)
 	}
+	parsed, err := aws.ParseRoleARN(integration.GetAWSOIDCIntegrationSpec().RoleARN)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
```

Comment:
The integration might not exist, and in that case `integration` will be `nil`, thus we can't get the RoleARN.
For those scenarios, we could return clearly fake account id (`000000000000`).
However, that would disclose that this integration does not exist in a public/no-auth endpoint.

Reply:
thanks for catching this, I didn't notice the not found error handling above.

I've reworked the PR to pass the account id via query param as you suggested, so the point is moot now.